### ISSUE TITLE
Unable to connect to this Samsung TV device (QE75Q80TAL)

### ISSUE THREAD

**Issue by sanderk86** [ollo69/ha-samsungtv-smart#79]
I cannot connect the component to my Samsung QE75Q80TAL.
I installed the component with HACS but when i then try to install the integration i am getting the error "Unable to connect to this Samsung TV device" (See the attached screenshot below)
I do get the permission popup on my TV and i do click allow.

![unabletoconnect](https://user-images.githubusercontent.com/5902123/118790554-2b89fa00-b896-11eb-8065-140e4a81405c.png)

When I go to http://192.168.10.164:8001/api/v2/ i am getting the following information:

_{"device":{"FrameTVSupport":"false","GamePadSupport":"true","ImeSyncedSupport":"true","Language":"nl_NL","OS":"Tizen","PowerState":"on","TokenAuthSupport":"true","VoiceSupport":"true","WallScreenRatio":"0","WallService":"false","countryCode":"NL","description":"Samsung DTV RCR","developerIP":"0.0.0.0","developerMode":"0","duid":"uuid:e876c0be-c666-4a47-9b85-d7cf99d0779b","firmwareVersion":"Unknown","id":"uuid:e876c0be-c666-4a47-9b85-d7cf99d0779b","ip":"192.168.10.164","model":"20_NIKEM_QTV","modelName":"QE75Q80TALXXN","name":"[TV] Samsung Q80 Series (75)","networkType":"wireless","resolution":"3840x2160","smartHubAgreement":"true","ssid":"f4:92:bf:a5:bf:f6","type":"Samsung SmartTV","udn":"uuid:e876c0be-c666-4a47-9b85-d7cf99d0779b","wifiMac":"BC:7E:8B:AA:5E:4E"},"id":"uuid:e876c0be-c666-4a47-9b85-d7cf99d0779b","isSupport":"{\"DMP_DRM_PLAYREADY\":\"false\",\"DMP_DRM_WIDEVINE\":\"false\",\"DMP_available\":\"true\",\"EDEN_available\":\"true\",\"FrameTVSupport\":\"false\",\"ImeSyncedSupport\":\"true\",\"TokenAuthSupport\":\"true\",\"remote_available\":\"true\",\"remote_fourDirections\":\"true\",\"remote_touchPad\":\"true\",\"remote_voiceControl\":\"true\"}\n","name":"[TV] Samsung Q80 Series (75)","remote":"1.0","type":"Samsung SmartTV","uri":"http://192.168.10.164:8001/api/v2/","version":"2.0.25"}_

Hope you can help me find a solution. Thanks in advance!

**Comment by ollo69** [comment]
You should have some details about connection tentative in HA log. Can you share this?

**Comment by sanderk86** [comment]
Thanks for the quick responds!

I have this in my HA log... or should I enable debug logging first?

"2021-05-19 11:26:39 ERROR (SyncWorker_0) [custom_components.samsungtv_smart] Web socket connection to SamsungTV 192.168.10.164 failed"

**Comment by sanderk86** [comment]
I enabled debug and got this in de log:

_2021-05-19 11:56:35 DEBUG (MainThread) [custom_components.samsungtv_smart.api.smartthings] SmartThings available devices: {'items': [{'deviceId': '30acd418-c88c-6356-7313-de92f3b47d0f', 'name': '[TV] Samsung Q80 Series (75)', 'label': 'Samsung Q80 Series (75)', 'manufacturerName': 'Samsung Electronics', 'presentationId': 'VD-STV_2018_K', 'deviceManufacturerCode': 'Samsung Electronics', 'locationId': '0fb4540f-78e6-479a-a6dc-eaef6eccb229', 'ownerId': '385b8f13-8864-cb65-3d37-5816ac982be2', 'roomId': '3e2f511c-04d8-44ea-99bb-f84c8077e13b', 'deviceTypeName': 'Samsung OCF TV', 'components': [{'id': 'main', 'capabilities': [{'id': 'ocf', 'version': 1}, {'id': 'switch', 'version': 1}, {'id': 'audioVolume', 'version': 1}, {'id': 'audioMute', 'version': 1}, {'id': 'tvChannel', 'version': 1}, {'id': 'mediaInputSource', 'version': 1}, {'id': 'mediaPlayback', 'version': 1}, {'id': 'mediaTrackControl', 'version': 1}, {'id': 'custom.error', 'version': 1}, {'id': 'custom.picturemode', 'version': 1}, {'id': 'custom.soundmode', 'version': 1}, {'id': 'custom.accessibility', 'version': 1}, {'id': 'custom.launchapp', 'version': 1}, {'id': 'custom.recording', 'version': 1}, {'id': 'custom.tvsearch', 'version': 1}, {'id': 'custom.disabledCapabilities', 'version': 1}, {'id': 'samsungvd.ambient', 'version': 1}, {'id': 'samsungvd.ambientContent', 'version': 1}, {'id': 'samsungvd.ambient18', 'version': 1}, {'id': 'samsungvd.mediaInputSource', 'version': 1}, {'id': 'refresh', 'version': 1}, {'id': 'execute', 'version': 1}, {'id': 'samsungvd.firmwareVersion', 'version': 1}, {'id': 'samsungvd.supportsPowerOnByOcf', 'version': 1}], 'categories': [{'name': 'Television', 'categoryType': 'manufacturer'}]}], 'createTime': '2021-02-21T19:09:36.235Z', 'profile': {'id': '272abfae-2be3-317c-810d-104576b73fe5'}, 'type': 'OCF', 'restrictionTier': 0}], '_links': {}}
2021-05-19 11:56:35 INFO (MainThread) [custom_components.samsungtv_smart.api.smartthings] SmartThings discovered TV devices: {'30acd418-c88c-6356-7313-de92f3b47d0f': {'name': '[TV] Samsung Q80 Series (75)', 'label': 'Samsung Q80 Series (75)'}}
2021-05-19 11:56:35 INFO (SyncWorker_1) [custom_components.samsungtv_smart] Try to configure SamsungTV 192.168.10.164 using port 8001
2021-05-19 11:56:35 DEBUG (SyncWorker_1) [custom_components.samsungtv_smart.api.samsungws] WS url ws://192.168.10.164:8001/api/v2/channels/samsung.remote.control?name=W0hvbWUgQXNzaXN0YW50XSBTYW1zdW5nIFE4MA%3D%3D
2021-05-19 11:56:35 DEBUG (SyncWorker_1) [custom_components.samsungtv_smart.api.samsungws] {'event': 'ms.channel.unauthorized'}
2021-05-19 11:56:35 INFO (SyncWorker_1) [custom_components.samsungtv_smart] Configuration failed using port 8001, error: {'event': 'ms.channel.unauthorized'}
2021-05-19 11:56:35 INFO (SyncWorker_1) [custom_components.samsungtv_smart] Try to configure SamsungTV 192.168.10.164 using port 8002
2021-05-19 11:56:35 DEBUG (SyncWorker_1) [custom_components.samsungtv_smart.api.samsungws] WS url wss://192.168.10.164:8002/api/v2/channels/samsung.remote.control?name=W0hvbWUgQXNzaXN0YW50XSBTYW1zdW5nIFE4MA%3D%3D
2021-05-19 11:56:41 DEBUG (SyncWorker_1) [custom_components.samsungtv_smart.api.samsungws] {'event': 'ms.remote.touchDisable'}
2021-05-19 11:56:41 INFO (SyncWorker_1) [custom_components.samsungtv_smart] Configuration failed using port 8002, error: {'event': 'ms.remote.touchDisable'}
2021-05-19 11:56:41 ERROR (SyncWorker_1) [custom_components.samsungtv_smart] Web socket connection to SamsungTV 192.168.10.164 failed_

**Comment by sanderk86** [comment]
No it is in the same VLAN.

I tried to go to reset on the TV but it was greyed out... I found somewhere online that if you have that there is a app running in the background. So I thought let me cut the power to the TV and restart it. After I done that I tried to connect te component again, and voila! It worked!

So maybe this is a good tip to mention in the configuration part :-)

Thanks for your time!
